apache/metron
# METRON-2023 [UI] Covering Grok Parser Creation with Cypress tests

## Contributor Comments

In this PR I introducing Cypress in Config UI and adding a Cypress test to cover Grok parser creation. I made a simplified common test setup across Config and Alert UI to bring them to the same platform.
The changeset might seem big, but most of the new files are contains configs or mock data.

The list of changes are the following:
- bumping Cypress version to 3.2.0 (it solve former issues yauzl/cypress binary unzipping)
- adding Cypress to Config UI
- using standard ng serve command to fire up both Alert and Config UI's for testing 
- adding Grok Parser creation test

"Full Dev" vm build tested and works fine.

## Steps to validate

Please check build logs in Travis CI to see the output of the [UI tests](https://travis-ci.org/tiborm/metron/jobs/510929760).

Under [this link](https://travis-ci.org/tiborm/metron/jobs/502000711) you can find a build failing where I intentionally broke a test to see whether it breaks the build too.


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
